shard-operator-parent/shard-operator-core/src/main/java/com/redhat/service/smartevents/shard/operator/core/observability/ObservabilitySetupService.java

```diff
@@ -0,0 +1,175 @@
+            LOGGER.error("One or more Observability parameters was undefined.");
+            logParameters();
+            // Mis-configuration is terminal as Observability must be running if enabled.
+            Quarkus.asyncExit(1);
+            return;
+        }
+
+        LOGGER.info("Observability is enabled.");
+        logParameters();
+
+        Secret existing = kubernetesClient
+                .secrets()
+                .inNamespace(namespace)
+                .withName(name)
+                .get();
+
+        Secret expected = templateProvider.loadObservabilitySecretTemplate(name, namespace);
+        expected.getData().put(OBSERVABILITY_ACCESS_TOKEN,
+                ENCODER.encodeToString(accessToken.getBytes(StandardCharsets.UTF_8)));
+        expected.getData().put(OBSERVABILITY_REPOSITORY,
+                ENCODER.encodeToString(repository.getBytes(StandardCharsets.UTF_8)));
+        expected.getData().put(OBSERVABILITY_CHANNEL,
+                ENCODER.encodeToString(channel.getBytes(StandardCharsets.UTF_8)));
+        expected.getData().put(OBSERVABILITY_TAG,
+                ENCODER.encodeToString(tag.getBytes(StandardCharsets.UTF_8)));
+
+        if (existing == null || !expected.getData().equals(existing.getData())) {
+            LOGGER.info("Existing Observability Secret either does not exist or differs to that required. Creating...");
+            try {
+                // Ensure the target Namespace exists
+                final Namespace ns = kubernetesClient.namespaces().withName(namespace).get();
+                if (ns == null) {
+                    LOGGER.info("Target Namespace for Observability Secret does not exist. Creating...");
+                    kubernetesClient.namespaces().createOrReplace(
+                            new NamespaceBuilder()
+                                    .withNewMetadata()
+                                    .withName(namespace)
+                                    .withLabels(new LabelsBuilder().buildWithDefaults(LabelsBuilder.V1_OPERATOR_NAME))
+                                    .endMetadata()
+                                    .build());
+                }
+
+                // Ensure the Secret exists
+                kubernetesClient
+                        .secrets()
+                        .inNamespace(expected.getMetadata().getNamespace())
+                        .withName(expected.getMetadata().getName())
+                        .createOrReplace(expected);
+            } catch (RuntimeException e) {
+                LOGGER.error("Failed to create Observability Secret. Please make sure it was properly configured.");
+                // Mis-configuration is terminal as Observability must be running if enabled.
+                Quarkus.asyncExit(1);
+            }
+        }
+    }
+
+    private void logParameters() {
+        LOGGER.debug("Name={}\n" +
+                "Namespace={}\n" +
+                "AccessToken={}\n" +
```

Comment:
@manstis Are we printing an access token in plaintext here?

Reply:
Erm, yes. Removed.